### PR TITLE
Less easy to check to be unquestionably good changes — Episode 1: liberation from X11

### PR DESCRIPTION
**Pull request by oxij**
Merge https://github.com/NixOS/nixpkgs/pull/6968 first, then the main change here is b04e26830e43c1aebc08f9b7d0ab41a08e3bf609, the ones before it (in a way git sorts stuff, not github) are here to make it work, the ones after it are just a bonus.

### CODE REVIEW

**Comment by jagajaga** [comment]
Maybe rebase and leave only unique commits? And write, that this PR needs https://github.com/NixOS/nixpkgs/pull/6968?

File: pkgs/os-specific/linux/conky/default.nix [hunk 1]
```diff
@@ -39,30 +67,47 @@ stdenv.mkDerivation rec {
...
-    ++ optionals weatherXoap  [ curl libxml2 ]
+    ++ optional  wirelessSupport    wirelesstools
+
+    ++ optional  curlSupport        curl
+    ++ optional  rssSupport         libxml2
+    ++ optional  weatherXoapSupport libxml2
     ;
 
   configureFlags =
     let flag = state: flags: if state then map (x: "--enable-${x}")  flags
```
Comment (jagajaga) [inline pkgs/os-specific/linux/conky/default.nix]: :+1:

**Comment by oxij** [comment]
@jagajaga not possible, it actually changes some lines changed in #6968

File: pkgs/development/tools/toluapp/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchFromGitHub {
+    owner = "eddieringle";
+    repo  = "toluapp";
+    rev   = "b1e680dc486c17128a3c21f89db1693ff06c02b1";
+    sha256 = "1d1a9bll9825dg4mz71vwykvfd3s5zi2yvzbfsvlr3qz1l3zqfwb";
+  };
+
+  buildInputs = [ lua scons ];
+
+  patches = [ /home/oxij/src/toluapp/oxij.patch ];
```
Comment (ardumont) [inline pkgs/development/tools/toluapp/default.nix]: Beware here your patch is local to your machine.

Comment (oxij) [inline pkgs/development/tools/toluapp/default.nix]: Yep, thanks. That's already fixed in the PR this was split into.

**Comment by oxij** [comment]
Closing this since #6968 didn't go as good as expected :)

**Pull request closed without merging**